src/openms/include/OpenMS/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.h

```diff
@@ -415,11 +443,45 @@ namespace OpenMS
       FeatureMap& features
     );
 
+    /**
+      @brief store MS1 and the associated MS2 features
+
+      @param[in] filename the filename of the file to write
+      @param[in] ms1_features the MS1 features
+      @param[in] ms2_features the MS2 features
+    */
+    void storeSpectraTraML(const String& filename, const OpenMS::FeatureMap& ms1_features, const OpenMS::FeatureMap& ms2_features) const;
+
+    /**
+      @brief store spectra in MSP format
+
+      @param[in] filename the filename of the file to write
+      @param[in] experiment the experiment to store
+    */
+    void storeSpectraMSP(const String& filename, MSExperiment& experiment) const;
+    
+    /**
+      @brief organize into a map by combining features and subordinates with the same `identifier`
+
+      @param[in] fmap_input input features map
+      @param[in] fmap_output output features map
+    */
+    void mergeFeatures(const OpenMS::FeatureMap& fmap_input, OpenMS::FeatureMap& fmap_output) const;
+
 protected:
     /// Overridden function from DefaultParamHandler to keep members up to date, when a parameter is changed
     void updateMembers_() override;
 
-private:
+    /// Deisotope MS2 spectra
+    void deisotopeMS2Spectra_(MSExperiment& experiment) const;
+
+    /// Remove peaks form MS2 which are at a higher mz than the precursor + 10 ppm
+    void removeMS2SpectraPeaks_(MSExperiment& experiment) const;
```

Comment:
We have to keep in mind/comment, that this only works correctly for compounds with charge 1.

Reply:
yes the charge 1 precursor m/z as reference should be fine.
But if we have a charge 1 precursor and you want to keep the isotope pattern then
you need to choose a larger tolerance

Reply:
I think adding a parameter is a good idea!